aiounifi/interfaces/api_handlers.py

```diff
@@ -98,14 +98,15 @@ def __init__(self, controller: "Controller") -> None:
     @final
     async def update(self) -> None:
         """Refresh data."""
-        raw = await self.controller.request(self.api_request)
-        self.process_raw(raw.get("data", []))
+        data = await self.controller.request(self.api_request)
+        self.process_raw(data)
 
     @final
-    def process_raw(self, raw: list[dict[str, Any]]) -> None:
+    def process_raw(self, raw: TypedApiResponse) -> None:
         """Process full raw response."""
-        for raw_item in raw:
-            self.process_item(raw_item)
+        if "data" in raw:
+            for raw_item in raw["data"]:
+                self.process_item(raw_item)
```

Comment:
process_raw should not handle variations, that's why its marked `@final` lets make sure that data us prepared to an expected format.

Reply:
That's correct, but without this extra check I saw some errors in the tests. But I agree that it shouldn't be needed to change the tests if done properly. I'm also struggling with this a bit.
Since we now have a TypedApiResponse, maybe another method is needed, because we are not processing a 'raw' response anymore. I need to put some more time and thought into this.

Reply:
Well you would pass the data field into process raw which would still be a list[dict]] right?